clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2581,16 +2581,17 @@ def SYCLIntelUseStallEnableClustersAttrDocs : Documentation {
   let Category = DocCatFunction;
   let Heading = "intel::use_stall_enable_clusters";
   let Content = [{
-When applied to a lambda or function call operator (of a function object)
-on device, this requests, to the extent possible, that statically-scheduled
-clusters handle stalls using a stall-enable signal to freeze computation
-within the cluster. This attribute is ignored on the host.
+When applied to a lambda function, function definition, or function call
+operator (of a function object) on device or host, this requests, to the
+extent possible, that statically-scheduled clusters handle stalls using a
+stall-enable signal to freeze computation within the cluster. This attribute
```

Comment:
Can you confirm with someone from the backend whether the attribute makes sense on host?

Reply:
waiting for conformation from @mendell27

Reply:
I checked with @mendell27 and here is response:
 It is not relevant to host mode, but it would be harmless.  No one else would look at the metadata

Reply:
Since allowing the attribute on host seems to be harmless, I did not make any change on Host mode. 

But I will leave this decision on reviewers @elizabethandrews or @AaronBallman or @erichkeane if this sounds OK.

Reply:
I don't particularly care here... I would think our best bet is to just 'accept' the attribute and make it be a no-op in IR-CodeGen for host-mode (that is, only generate the metadata in device-mode, but have everything else the same).

The only reason I would see to just make it work the same in both is to simplify writing the tests (Since you can do -fsycl-is-host do have the sycl logic without having to compel emission).

Reply:
Thanks @erichkeane,

>> I would think our best bet is to just 'accept' the attribute and make it be a no-op in IR-CodeGen for host-mode (that is, only generate the metadata in device-mode, but have everything else the same).

I have updated PR to generate the the metadata only in device-mode.

Reply:
Does this attribute have an effect outside of SYCL? If not, we should mention that in the docs -- otherwise, the mentions of device and host are a bit strange.

Reply:
>>Does this attribute have an effect outside of SYCL? If

We use GNU spelling (for OpenCL) for same attribute in downstream. 

https://www.intel.com/content/dam/www/programmable/us/en/pdfs/literature/hb/opencl-sdk/aocl_programming_guide.pdf

Reply:
But we don't seem to in syclos (unless I'm missing something), so we should probably mention in the docs that this attribute requires SYCL so that mentions of "host" and "device" have some context.

Reply:
>  so we should probably mention in the docs that this attribute requires SYCL so that mentions of "host" and "device" have some context.

Agreed. I have updated Doc.